src/meshapi/views/forms.py

```diff
@@ -191,13 +215,37 @@ def join_form(request):
     )
 
 
+def get_next_free_nn() -> int:
+    # We don't need to lock all the DB objects, just the ones that impact the free NN calculations
```

Comment:
Could we get away with just the first one, join form style?

Reply:
> Just the ones that impact the free NN calculations
Do you have an example?

> Just the first one
Not totally clear on what locking the first one does.